src/analysis/inner_product_space/linear_pmap.lean

```diff
@@ -0,0 +1,228 @@
+domain of `S`, we have that `⟪T x, y⟫ = ⟪x, S y⟫`. -/
+def is_formal_adjoint (T : E →ₗ.[𝕜] F) (S : F →ₗ.[𝕜] E) : Prop :=
+∀ (x : T.domain) (y : S.domain), ⟪T x, y⟫ = ⟪(x : E), S y⟫
+
+variables {T : E →ₗ.[𝕜] F} {S : F →ₗ.[𝕜] E}
+
+@[protected] lemma is_formal_adjoint.symm (h : T.is_formal_adjoint S) : S.is_formal_adjoint T :=
+λ y _, by rw [←inner_conj_symm, ←inner_conj_symm (y : F), h]
+
+variables (T)
+
+/-- The domain of the adjoint operator.
+
+This definition is needed to construct the adjoint operator and the preferred version to use is
+`T.adjoint.domain` instead of `T.adjoint_domain`. -/
+def adjoint_domain : submodule 𝕜 F :=
+{ carrier := {y | continuous ((innerₛₗ 𝕜 y).comp T.to_fun)},
+  zero_mem' := by { rw [set.mem_set_of_eq, linear_map.map_zero, linear_map.zero_comp],
+      exact continuous_zero },
+  add_mem' := λ x y hx hy, by { rw [set.mem_set_of_eq, linear_map.map_add] at *, exact hx.add hy },
+  smul_mem' := λ a x hx, by { rw [set.mem_set_of_eq, linear_map.map_smulₛₗ] at *,
+    exact hx.const_smul (conj a) } }
+
+/-- The operator `λ x, ⟪y, T x⟫` considered as a continuous linear operator from `T.adjoint_domain`
+to `𝕜`. -/
+def adjoint_domain_mk_clm (y : T.adjoint_domain) : T.domain →L[𝕜] 𝕜 :=
+⟨(innerₛₗ 𝕜 (y : F)).comp T.to_fun, y.prop⟩
+
+lemma adjoint_domain_mk_clm_apply (y : T.adjoint_domain) (x : T.domain) :
+  adjoint_domain_mk_clm T y x = ⟪(y : F), T x⟫ := rfl
+
+variable {T}
+variable (hT : dense (T.domain : set E))
+
+include hT
+
+/-- The unique continuous extension of the operator `adjoint_domain_mk_clm` to `E`. -/
+def adjoint_domain_mk_clm_extend (y : T.adjoint_domain) :
+  E →L[𝕜] 𝕜 :=
+(T.adjoint_domain_mk_clm y).extend (submodule.subtypeL T.domain)
+  hT.dense_range_coe uniform_embedding_subtype_coe.to_uniform_inducing
+
+@[simp] lemma adjoint_domain_mk_clm_extend_apply (y : T.adjoint_domain) (x : T.domain) :
+  adjoint_domain_mk_clm_extend hT y (x : E) = ⟪(y : F), T x⟫ :=
+continuous_linear_map.extend_eq _ _ _ _ _
+
+variables [complete_space E]
+
+lemma exists_unique_adjoint_elem (y : T.adjoint_domain) : ∃! (w : E),
+  ∀ (x : T.domain), ⟪w, x⟫ = ⟪(y : F), T x⟫ :=
+exists_unique_of_exists_of_unique
+  -- For the existence we use the Fréchet-Riesz representation theorem and extend
+  -- the map that is only defined on `T.domain` to `E`:
+  ⟨(inner_product_space.to_dual 𝕜 E).symm (adjoint_domain_mk_clm_extend hT y),
+    -- Implementation note: this is true `by simp`
+    by simp only [inner_product_space.to_dual_symm_apply, adjoint_domain_mk_clm_extend_apply hT,
+      eq_self_iff_true, forall_const]⟩
+  -- The uniqueness follows directly from the fact that `T.domain` is dense in `E`.
+  (λ _ _ hy₁ hy₂, hT.eq_of_inner_left (λ v, (hy₁ v).trans (hy₂ v).symm))
+
+/-- The image of the adjoint operator.
+
+This is an auxiliary definition needed to define the adjoint operator as a `linear_pmap`. -/
+def adjoint_elem (y : T.adjoint_domain) : E :=
+(exists_unique_adjoint_elem hT y).exists.some
```

Comment:
If I'm reading the above correctly, can this be defined as
```suggestion
(inner_product_space.to_dual 𝕜 E).symm (adjoint_domain_mk_clm_extend hT y)
```
Using a witness to construct a unique existential, then using `some` to get it back out again seems a bit convoluted.

Reply:
You are completely right, this was overly complicated. Thanks!